controllers/gatekeeper_controller.go

```diff
@@ -51,3 +119,60 @@ func (r *GatekeeperReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&operatorv1alpha1.Gatekeeper{}).
 		Complete(r)
 }
+
+func (r *GatekeeperReconciler) deployGatekeeperResources() error {
+	ctx := context.Background()
+	var err error
+	for _, a := range orderedStaticAssets {
+		assetName := staticAssetsDir + a
+		bytes, err := bindata.Asset(assetName)
+		if err != nil {
+			return err
+		}
+
+		logger := r.Log.WithValues("Gatekeeper resource", string(assetName))
+		manifest := &manifest.Manifest{}
+		err = manifest.UnmarshalJSON(bytes)
+		if err != nil {
+			return err
+		}
+
+		err = r.Create(ctx, manifest.Obj)
+		if err == nil {
+			logger.Info(fmt.Sprintf("Created Gatekeeper resource"))
+			continue
+		}
+
+		// Create returned an error, now process it.
+
+		if !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+
+		clusterObj := &unstructured.Unstructured{}
```

Comment:
It took a bit to understand what's happening here. Maybe (and just maybe) it would be clearer the other way around, trying to get the object, and creating if it does not exists.

The current flow works because err is not nil and because is equal to already exists, but it's not totally clear.

Reply:
@fedepaol I've pushed an update with some refactoring. Let me know if this is any better to follow.

Reply:
Looks way better, thanks for taking the time to do it!